### ISSUE TITLE
[Bug]: [scale] Standalone load fails  and flush hangs after oomkilled and scaling-up restart

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#18652]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.0-20220812-bc62ca1f
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.0.dev6
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

argo-workflow name: `scale-up-oom-sk85w`
1. Deploy standalone and limit resources
```
resources:
    requests:
      memory: 128Mi
      cpu: 100m
    limits:
      memory: 9Gi
      cpu: 2
```
standalone pod:
```
standalone-timeout-milvus-standalone-79887ffc9d-mgmc6   1/1     Running   2 (76s ago)   11m
```
2. Run the case
```
    def test_insert_oom(self, host):
        default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}

        # collection fields and schema
        fields = [cf.gen_int64_field(is_primary=True), cf.gen_double_field(), cf.gen_float_vec_field(dim=3840)]

        def do_collection():
            """ create collection, insert, load, search, query"""
            schema, _ = ApiCollectionSchemaWrapper().init_collection_schema(fields=fields, auto_id=True)
            collection_w = ApiCollectionWrapper()
            collection_w.init_collection(name=cf.gen_unique_str("oom_"), schema=schema, shards_num=16)
            for i in range(10):
                df = pd.DataFrame({
                    ct.default_double_field_name: pd.Series(data=[np.double(i) for i in range(0, 2000)],
                                                            dtype="double"),
                    ct.default_float_vec_field_name: cf.gen_vectors(2000, dim=3840)
                })
                insert_res, _ = collection_w.insert(df, timeout=180, check_task=CheckTasks.check_nothing)
            # log.debug(collection_w.num_entities)

            collection_w.load(timeout=120)
            log.debug(collection_w.get_replicas(), check_task=CheckTasks.check_nothing)
            query_res, _ = collection_w.query(expr=f"{ct.default_int64_field_name} in {insert_res.primary_keys[0:10]}", timeout=120)
            assert len(query_res) == 10
            search_res, _ = collection_w.search(cf.gen_vectors(2, dim=3840), ct.default_float_vec_field_name,
                                                default_search_params, 10, timeout_decorator=120)
            assert len(search_res) == 2
            assert len(search_res[0]) == 10

        try:
            tasks = []
            with ThreadPoolExecutor(max_workers=8) as t:
                for i in range(20):
                    task = t.submit(do_collection)
                    tasks.append(task)

            for task in tasks:
                task.done()
        except Exception as e:
            log.error(str(e))
```
3. Standalone pod oomkilled and restart sometimes
```
standalone-timeout-milvus-standalone-6c5b5b7478-vxdzf   1/1     Running   0          23s
```
4. Scale-up cpu and memory
```
resources:
    limits:
      memory: 16Gi
      cpu: 4
```
5. Wait pods ready and run the check case

- e2e-test check, load collection failed
```
[2022-08-15 05:14:32 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 20} (api_request.py:56)
[2022-08-15 05:14:32 - DEBUG - ci_test]: (api_response) : None  (api_request.py:31)
[2022-08-15 05:14:32 - INFO - ci_test]: [test][2022-08-15T05:14:32Z] [0.05926175s] e2e__ySI564pk flush -> None (wrapper.py:30)
[2022-08-15 05:14:32 - INFO - ci_test]: assert flush: 4.096614360809326, entities: 3000 (test_e2e.py:41)
[2022-08-15 05:14:32 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:56)
[2022-08-15 05:14:52 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=rpc deadline e
xceeded: Retry timeout: 20s)>, <Time:{'RPC start': '2022-08-15 05:14:32.762355', 'RPC error': '2022-08-15 05:14:52.763394'}> (decora
tors.py:95)
[2022-08-15 05:14:52 - ERROR - ci_test]: Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 48, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 673, in load_collection
    response = rf.result()
  File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.DEADLINE_EXCEEDED
        details = "Deadline Exceeded"
        debug_error_string = "{"created":"@1660540492.763138120","description":"Deadline Exceeded","file":"src/core/ext/filters/dead
line/deadline_filter.cc","file_line":81,"grpc_status":4}"
>
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/src/milvus/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/src/milvus/tests/python_client/utils/api_request.py", line 57, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 474, in load
    conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 96, in handler
    raise e
```
- oom-collection check flush hangs:
```
[2022-08-15 05:15:06,963 - DEBUG - ci_test]: (api_request)  : [Collection] args: ['oom__TwaJZopL', None, 'default', 2], kwargs: {'co
nsistency_level': 'Strong'} (api_request.py:56)
[2022-08-15 05:15:06,967 - DEBUG - ci_test]: (api_response) : <Collection>:
-------------
<name>: oom__TwaJZopL
<partitions>: [{"name": "_default", "collection_name": "oom__TwaJZopL", "description": ""}]
<description>:
<schema>: {
  auto_id: True
  description:
  fields: [{
    name: int64
    description:
    type: 5
    is_primary: True
    auto_id: True
......  (api_request.py:31)
[2022-08-15 05:15:06,967 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 20} (api_request.py:5
6)
Error: signal: terminated
```

**Comment by soothing-rain** [comment]
Any server side logs?

**Comment by ThreadDao** [comment]
> Any server side logs?

Go to Loki ?

**Comment by ThreadDao** [comment]
@congqixia There also has a problem with load failure. Are you free to help take a look?
1. Deploy standalone with CPU 2 and memory 9Gi limit
2. Concurrent insert data into 20 collections which has 16 shards, and pod OOMKilled
3. Scale up standalone CPU to 4, memory to 16Gi, and pod re-running
4. Run the e2e-test, and collection always load failed
```
[2022-08-19 09:56:48 - DEBUG - ci_test]: (api_request)  : [Collection.flush] args: [], kwargs: {'timeout': 60} (api_request.py:56)
[2022-08-19 09:56:48 - DEBUG - ci_test]: (api_response) : None  (api_request.py:31)
[2022-08-19 09:56:48 - INFO - ci_test]: [test][2022-08-19T09:56:48Z] [0.00164643s] e2e__FfCV3O2c flush -> None (wrapper.py:30)
[2022-08-19 09:56:48 - INFO - ci_test]: assert flush: 5.6184117794036865, entities: 3000 (test_e2e.py:41)
[2022-08-19 09:56:48 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:56)
[2022-08-19 09:57:08 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)>, <Time:{'RPC start': '2022-08-19 09:56:48.389442', 'RPC error': '2022-08-19 09:57:08.390622'}> (decorators.py:95)
[2022-08-19 09:57:08 - ERROR - ci_test]: Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 48, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 674, in load_collection
    response = rf.result()
  File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1660903028.390118736","description":"Deadline Exceeded","file":"src/core/ext/filters/deadline/deadline_filter.cc","file_line":81,"grpc_status":4}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/src/milvus/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/src/milvus/tests/python_client/utils/api_request.py", line 57, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 474, in load
    conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 96, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 92, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 62, in handler
    raise MilvusException(Status.UNEXPECTED_ERROR, f"rpc deadline exceeded: {timeout_msg}")
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)>
 (api_request.py:39)
[2022-08-19 09:57:08 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)> (api_request.py:40)
```

client pod: scale-up-oom-tcgts-3661422963, cluster: devops, namespace: qa
milvus pod before oom: standalone-soothing-milvus-standalone-7969cb78b8-vqvbf, cluster: devops, namespace: chaos-testing
milvus pod after oom: standalone-soothing-milvus-standalone-7454b5b9cb-hvkkf, cluster: devops, namespace: chaos-testing

**Comment by soothing-rain** [comment]
Flush hang issues is resolved in https://github.com/milvus-io/milvus/pull/18733

@congqixia Can you help with the load failure issue? Thanks!

/unassign
/assign @congqixia

**Comment by jiaoew1991** [comment]
> Flush hang issues is resolved in #18733
> 
> @congqixia Can you help with the load failure issue? Thanks!
> 
> /unassign /assign @congqixia

Does load failure bug like this one https://github.com/milvus-io/milvus/issues/18664 ?

**Comment by soothing-rain** [comment]
> 

Could chat with @ThreadDao if there's an image with a fix that we'd like to try.

**Comment by soothing-rain** [comment]
> > Flush hang issues is resolved in #18733
> > @congqixia Can you help with the load failure issue? Thanks!
> > /unassign /assign @congqixia
> 
> Does load failure bug like this one #18664 ?


Could chat with @ThreadDao if there's an image with a fix that we'd like to try.

**Comment by ThreadDao** [comment]
@weiliu1031 No fix
image: `2.1.0-20220824-85bbba95`
client pod: scale-up-oom-ldmv2-4185682159, clusetr: devops,  namespace: qa
```
[2022-08-24 03:59:15 - INFO - ci_test]: [test][2022-08-24T03:59:15Z] [0.00148648s] e2e__OGE35K8F flush -> None (wrapper.py:30)
[2022-08-24 03:59:15 - INFO - ci_test]: assert flush: 3.5384769439697266, entities: 3000 (test_e2e.py:41)
[2022-08-24 03:59:15 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 20], kwargs: {} (api_request.py:56)
[2022-08-24 03:59:35 - ERROR - pymilvus.decorators]: RPC error: [load_collection], <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)>, <Time:{'RPC start': '2022-08-24 03:59:15.699181', 'RPC error': '2022-08-24 03:59:35.700148'}> (decorators.py:95)
[2022-08-24 03:59:35 - ERROR - ci_test]: Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 48, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 674, in load_collection
    response = rf.result()
  File "/usr/local/lib/python3.8/dist-packages/grpc/_channel.py", line 744, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1661313575.699858332","description":"Deadline Exceeded","file":"src/core/ext/filters/deadline/deadline_filter.cc","file_line":81,"grpc_status":4}"
>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/src/milvus/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/src/milvus/tests/python_client/utils/api_request.py", line 57, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 474, in load
    conn.load_collection(self._name, replica_number=replica_number, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 96, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 92, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 62, in handler
    raise MilvusException(Status.UNEXPECTED_ERROR, f"rpc deadline exceeded: {timeout_msg}")
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)>
 (api_request.py:39)
[2022-08-24 03:59:35 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=rpc deadline exceeded: Retry timeout: 20s)> (api_request.py:40)
```
server pod before oom: standalone-oom-milvus-standalone-f68d6c77f-vhfng, cluster: devops, namespace: chaos-testing
server pod after oom: cluster: standalone-oom-milvus-standalone-bcdb4fbc9-x42jl, devops, namespace: chaos-testing

**Comment by weiliu1031** [comment]
> standalone-oom-milvus-standalone-f68d6c77f-vhfng
it's another problem, not relate to #18664

cause:
1. query coord panic
2. query coord is scheduling lots of handoff task after recover
![image](https://user-images.githubusercontent.com/108661493/186361455-f5a30777-2f3a-441a-8fb9-24af280055b1.png)

3. query coord receive a load collection task, and add to task queue
![image](https://user-images.githubusercontent.com/108661493/186361709-c283768e-d8d7-43cd-be9d-e9325cda92fd.png)

4. the load collection task has no chance to trigger.
![image](https://user-images.githubusercontent.com/108661493/186362117-bfc755e0-44a0-4440-af18-c568efbab18f.png)

**Comment by ThreadDao** [comment]
Sadly I used a timeout of 15h to load the 3000 entities and it still timed out

**Comment by weiliu1031** [comment]
more info about load collection task doesn't trigger.

1. in standalone, query coord trigger two NodeDown load balanceTask
2. the first one generate tasks, and finished peacefully
3. the second one also generate same tasks, such as WatchDmChannelTask, and send it to query node
4. this watchDmChannelTask failed due to dml channel has already loaded.
5. query coord retry this watchDmChannelTask on same node for 5 times, all failed with same reason.
6. query coord retry to reschedule this channel on another node, so it try to exclude the current query node, and assign this task to another node.
7. in assign logic, exclude current query node, no node available to assign. so it will wait until available node appear.
8. due to the only query node has been excluded, so wait forever.

**Comment by ThreadDao** [comment]
Fixed `2.1.0-20220830-0733aec`
